aptos-move/parallel-executor/src/proptest_types/tests.rs

```diff
@@ -212,3 +211,134 @@ fn module_publishing_fallback() {
         (true, true),
     ));
 }
+
+fn publishing_fixed_params() {
+    let mut runner = TestRunner::default();
+    let num_txns = 300;
+
+    let universe = vec(any::<[u8; 32]>(), 50)
+        .new_tree(&mut runner)
+        .expect("creating a new value should succeed")
+        .current();
+    let transaction_gen = vec(
+        any_with::<TransactionGen<[u8; 32]>>(TransactionGenParams::new_dynamic()),
+        num_txns,
+    )
+    .new_tree(&mut runner)
+    .expect("creating a new value should succeed")
+    .current();
+    let indices = vec(any::<Index>(), 4)
+        .new_tree(&mut runner)
+        .expect("creating a new value should succeed")
+        .current();
+
+    // First 12 keys are normal paths, next 14 are module reads, then writes.
+    let mut transactions: Vec<_> = transaction_gen
+        .into_iter()
+        .map(|txn_gen| txn_gen.materialize_disjoint_module_rw(&universe[0..40], 12, 26))
+        .collect();
+
+    // Adjust the writes of txn indices[0] to contain module write to key 42.
+    let w_index = indices[0].index(num_txns);
+    *transactions.get_mut(w_index).unwrap() = match &*transactions.get_mut(w_index).unwrap() {
+        Transaction::Write {
+            incarnation,
+            reads,
+            writes,
+        } => {
+            let mut new_writes = vec![];
+            for incarnation_writes in writes {
+                assert!(!incarnation_writes.is_empty());
+                let val = incarnation_writes[0].1;
+                let insert_idx = indices[1].index(incarnation_writes.len());
+                new_writes.push(
+                    incarnation_writes
+                        .iter()
+                        .enumerate()
+                        .flat_map(|(i, w)| {
+                            if i == insert_idx {
+                                vec![*w, (KeyType(universe[42], true), val)]
+                            } else {
+                                vec![*w]
+                            }
```

Comment:
this looks very weird, you just want `incarnation_writes.insert(insert_idx, (KeyType(universe[42, true), val))`